hedera-node/src/main/java/com/hedera/services/txns/contract/ContractCallTransitionLogic.java

```diff
@@ -164,6 +181,52 @@ public void doStateTransitionOperation(
 		recordService.externaliseEvmCallTransaction(result);
 	}
 
+	private TransactionProcessingResult constructResultLocally(Account receiver, ContractCallTransactionBody op,
+															   Pair<Long, Bytes> htsCallResult, HederaWorldState.Updater updater) {
+		final var output = htsCallResult.getValue();
+		//Any idea how to evaluate the status, since there are is no messageFrame's states?
+		var isSuccessful = !output.isEmpty() && !output.isZero();
+		final var requiredGas = htsCallResult.getKey();
+		final var gasPrice = evmTxProcessor.gasPriceTinyBarsGiven(txnCtx.consensusTime(), false);
+		final var intrinsicGas = evmTxProcessor.getIntrinsicGasCost(Bytes.EMPTY, false);
+		final var gasAvailable = op.getGas() - intrinsicGas;
+
+		final Map<Address, Map<Bytes, Pair<Bytes, Bytes>>> stateChanges;
+		//How to get the used gas?
+		if (properties.shouldEnableTraceability()) {
+			stateChanges = updater.getFinalStateChanges();
+		} else {
+			stateChanges = Map.of();
+		}
+		if (gasAvailable < requiredGas) {
+			//A custom exception or ransactionProcessingResult.failed?
+			isSuccessful = false;
+		}
+		final var gasUsedByTransaction = evmTxProcessor.calculateGasUsedByTokenCallTX(op.getGas(), gasAvailable);
```

Comment:
We should add logic for refunding accounts for both regular calls and eth calls, as it's done [here](https://github.com/hashgraph/hedera-services/blob/8f12fff2943bbbde02233155bc74156723a59e14/hedera-node/src/main/java/com/hedera/services/contracts/execution/EvmTxProcessor.java#L292)